11001.md

```diff
@@ -0,0 +1,90 @@
+NIP-XXXX
+======
+
+messages specification for peer 2 peer
+--------------------------------------
+
+`draft` `optional` `author:XXXX`
+
+Peer-to-peer is one of the most important applications needed in the Bitcoin space.
+
+Nostr can enhance liquidity by enabling existing profiles to use their reputation to trade with other peers. Reputation is a key element that makes large P2P platforms like Paxful or Binance P2P easy to use and scale, because it's easy to find someone who is likely reliable. The drawback is that identity in those platforms is locked in, and reputation built on one of those platforms can't be reused on another. Using Nostr would solve that reputation problem.
+
+Users could also choose to create a one-time identity and participate on the same based platform by simply creating a new key-pair for each trade. It would make it possible to merge currently separate use cases where platforms today are either relying on bonds (like RoboSats) or reputation (like Paxful, Binance P2P, HodlHodl).
+
+An ecosystem where different clients use the same specification on how to create and settle trades would create a thriving environment where clients compete in their UX and product offerings, to the benefit of the users and the global liquidity of the network.
+
+While it would be likely that clients are initially dedicated to this application, an ideal scenario would be that, over time, generic clients also integrate this NIP. This would incentivize a large group of users who would own Bitcoin, from zaps for instance, to trade with each other for local fiat currencies, without having the current barrier to entry of having a dedicated app.
+
+## Data structure
+
+an event looks like this:
+
+```
+let event = {
+  kind: 11001,
```

Comment:
Consider using nip33 for this, events with kind `30000 <= n < 40000` can be replaced by new ones, this way you can update the tags on the order in case you needed

Reply:
nip16: all events you send will be replaced, if you want to publish 2 orders, you publish the first one, then publish the second one, the first one will be replaced with the second one.

nip33: you identify the events with a specific `d` tag, so when you need to replace an event you just use that same `d` tag value.

Reply:
> nip16: all events you send will be replaced, if you want to publish 2 orders, you publish the first one, then publish the second one, the first one will be replaced with the second one.

gotcha. when you said `all events you send will be replaced`, the `you` represent an `npub`? ie: this is like a "singleton" instance per npub?

Reply:
yes, all events from the same npub will be replaced, so one user can't send more than one

Reply:
updated: https://github.com/nostr-protocol/nips/pull/405/commits/8713cf25e75fe2a4a5ee02a8d27394826ff4a373